e2e/test/helpers/TestDeviceCallbackHandler.cs

```diff
@@ -62,7 +62,7 @@ public async Task SetDeviceReceiveMethodAsync(string methodName, string deviceRe
                         var response = new Client.DirectMethodResponse()
                         {
                             Status = 200,
-                            Payload = Encoding.UTF8.GetBytes(deviceResponseJson)
```

Comment:
I'd like to see a few e2e tests that demonstrate sending and receiving custom JSON objects (type T) so that we know for sure there isn't any unexpected escaping of payloads that is done by the service